fvm/src/executor/default.rs

```diff
@@ -489,6 +443,8 @@ where
         self.state_tree_mut().mutate_actor(sender_id, |act| {
             act.deduct_funds(&gas_cost)?;
             act.sequence += 1;
+            // In case we've transformed an Embryo into an EthAccount
+            act.code = sender_state.code;
             Ok(())
         })?;
```

Comment:
This is a case where it's probably safe to just set instead of mutating. Or, even better, turn this entire block into one big mutation?